docs/compatibility.md

```diff
@@ -0,0 +1,69 @@
+
+
+## Synopsis
+*If a Manifest is pulled by _Digest_ from a Regsitry 2.3 with Docker Engine 1.9
+and older, and the Manifest was pushed with Docker Engine 1.10, a security check
+will cause the Engine to receive a Manifest it cannot use and the pull will fail.*
+
+## Registry Manifest Support
+
+
+Historically, the Registry has supported a [single Manifest type](https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-1.md).
+
+With the move toward multiple architecture images the distribution project
+introduced two new Manifest types:  Schema 2 Manifests and Manifest Lists.  The
+Registry 2.3 supports all three Manifest types and in order to be compatible
+with older Docker engines will, in certain cases, do an on-the-fly
+transformation of a Manifest before serving the JSON in the response.
+
+This conversion has some implications for pulling Manifests by Digest and this
+document enumerate these implications.
+
+
+## Content Adressable Storage (CAS)
+
+Manifests are stored and retreived in the Registry by keying off a Digest
+representing a hash of the contents.  One of the advantages provided by CAS is
+security: if the contents are changed, then the Digest will no longer match.
+This prevents any modification of the Manifest by MITM attack or an untrusted
+third party.
+
+When a Manifest is stored by the Registry, this Digest is returned in the HTTP
+response headers and, if events are configured, delivered within the event.  The
+Manifest can either be retreived by the Tag, or this Digest.
+
+For Registry versions 2.2.1 and below, the Registry will always store and
+serve _Schema 2 Version 1_ Manifests.  The Docker Engine 1.10 will first
+attempt to send a _Schema 2 version 2_ Manifest, falling back to sending a
+Schema 2 version 1 type Manifest when it detects that the Registry does not
+support the new version.
+
+
+## Registry v2.3
+
+### Manifest Push with Docker 1.9 and older
+
+The docker engine will construct a _Schama 2 Version 1_ Manifest which the
+Registry will persist to disk.
+
+When the Manifest is pulled by Digest or Tag with any Docker Version, a _Schema
+2 Version 1_ Manifest will be returned.
+
+### Manifest Push with Docker 1.10
+
+The docker engine will construct a _Schama 2 Version 2_ Manifest which the
+Registry will persist to disk.
+
+When the Manifest is pulled by Digest or Tag with Docker Engine 1.10, a
+_Schama 2 Version 2_ Manifest will be returned.  The Docker Engine 1.10
+understands the new Manifest format.
+
+When the Manifest is pulled by *Tag* with Docker Engine 1.9 and older, the
+Manifest is converted on-the-fly to _Scheam 2 Version 1_ and sent in the
+response.  The Docker Engine 1.9 is compatible with this older format.
+
+*When the Manifest is pulled by _Digest_ with Docker Engine 1.9 and older, the
+same rewriting process cannot happen in the Registry.  If this were to happen
+the Digest would no longer match the hash of the Manifest and would violate the
+constraints of CAS.*
```

Comment:
I suggest adding a sentence at the end concluding that this is why Docker 1.9 can't pull a manifest by digest that was pushed to Registry 2.3 by Docker 1.10+. This restates what you said in the synopsis, but it makes it absolutely clear that this setup doesn't work, and why not.